src/wr_control_drive_science/src/TeleopScienceControl.cpp

```diff
@@ -12,30 +12,27 @@
 #include <array>
 
 constexpr std::uint32_t MESSAGE_CACHE_SIZE = 10;
+constexpr int SAMPLES_PER_READING = 3;
 constexpr float TIMER_CALLBACK_DURATION = 5;
 
 ros::Publisher moistureSensor1;
-ros::Publisher moistureSensor2;
-ros::Publisher moistureSensor3;
 int baudRate;
 std::string file;
 int fd;
 
 void moistureCallback(const ros::TimerEvent &timerEvent) {
-    std::array<float, 3> buf{0, 0, 0};
+    std::array<float, SAMPLES_PER_READING> buf;
     /* Flush anything already in the serial buffer */
     tcflush(fd, TCIFLUSH);
     /* read up to 128 bytes from the fd */
-    int readStatus = read(fd, buf.data(), 3*sizeof(float));
+    int readStatus = read(fd, buf.data(), SAMPLES_PER_READING*sizeof(float));
     //NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay, cppcoreguidelines-pro-type-vararg, hicpp-no-array-decay, hicpp-no-assembler, hicpp-vararg)
     ROS_ASSERT_MSG(readStatus != -1, "Read failed");
     std_msgs::Float32 temp;
-    temp.data = buf[0];
-    moistureSensor1.publish(temp);
-    temp.data = buf[1];
-    moistureSensor1.publish(temp);
-    temp.data = buf[2];
-    moistureSensor1.publish(temp);
+    for(int i = 0; i < SAMPLES_PER_READING; i++) {
+        temp.data = buf[i];
+        moistureSensor1.publish(temp);
+    }
```

Comment:
Publishing all datapoints over `moistureSensor1` may still be malforming the data (how will we know which measurements are which?) and enumerating the sensor `Publisher`s explicitly may violate the dynamic behavior desired by using a `SAMPLES_PER_READING` variable.  Maybe it could be achieved via an `std::array` of `Publisher`s.  Then, publishing over that loop could be achieved via a range-based `for`-loop (we don't really care about the index of the data, just the correspondence).